provider/core/src/hello_world.rs

```diff
@@ -238,6 +238,31 @@ impl HelloWorldFormatter {
 
     crate::gen_any_buffer_constructors!(locale: include, options: skip, error: DataError);
 
+    /// Creates a new [`HelloWorldFormatter`] with global data.
+    ///
+    /// Requires the "globaldata" Cargo feature.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_provider::hello_world::HelloWorldFormatter;
+    /// use icu_locid::locale;
+    ///
+    /// let formatter = HelloWorldFormatter::try_new(&locale!("ja").into()).unwrap();
+    ///
+    /// assert_eq!(
+    ///     "こんにちは世界",
+    ///     formatter.format_to_string()
+    /// );
+    /// ```
+    #[cfg(feature = "globaldata")]
+    pub fn try_new(locale: &DataLocale) -> Result<Self, DataError> {
+        struct LocalBakedProvider;
+        use crate as icu_provider;
+        globaldata::impl_core_helloworld_v1!(LocalBakedProvider);
```

Comment:
An open question is how to enable vertical fallback. An easy way, which is probably fine, is

```rust
let provider = globaldata::with_vertical_fallback!(LocalBakedProvider);
```

and the macro has two versions: one that is a no-op, and one that is enabled with the `globaldata/fallback` feature that actually enables the fallback